sqlmesh/core/model/definition.py

```diff
@@ -2682,30 +2683,39 @@ def render_field_value(value: t.Any) -> t.Any:
 
     for field_name, field_info in ModelMeta.all_field_infos().items():
         field = field_info.alias or field_name
-        if field not in RUNTIME_RENDERED_MODEL_FIELDS and (field_value := fields.get(field)):
-            if isinstance(field_value, dict):
-                rendered_dict = {}
-                for key, value in field_value.items():
-                    if key in RUNTIME_RENDERED_MODEL_FIELDS:
-                        rendered_dict[key] = value
-                    elif rendered := render_field_value(value):
-                        rendered_dict[key] = rendered
-                if rendered_dict:
-                    fields[field] = rendered_dict
-                else:
-                    fields.pop(field)
-            elif isinstance(field_value, list):
-                if rendered_list := [
-                    rendered for value in field_value if (rendered := render_field_value(value))
-                ]:
-                    fields[field] = rendered_list
-                else:
-                    fields.pop(field)
+
+        if field in RUNTIME_RENDERED_MODEL_FIELDS:
+            continue
+
+        field_value = fields.get(field)
+        if field_value is None:
+            continue
+
+        if isinstance(field_value, dict):
+            rendered_dict = {}
+            for key, value in field_value.items():
+                if key in RUNTIME_RENDERED_MODEL_FIELDS:
+                    rendered_dict[key] = value
+                elif (rendered := render_field_value(value)) is not None:
+                    rendered_dict[key] = rendered
+            if rendered_dict:
+                fields[field] = rendered_dict
             else:
-                if rendered_field := render_field_value(field_value):
-                    fields[field] = rendered_field
-                else:
-                    fields.pop(field)
+                fields.pop(field)
+        elif isinstance(field_value, list):
+            rendered_list = [
+                rendered for value in field_value if (rendered := render_field_value(value))
+            ]
```

Comment:
Should we be checking for `None` within the comprehension as well?

Reply:
good catch, thank you!